rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py

```diff
@@ -153,157 +356,137 @@ def _map_features_to_indices(
-                f" Feature is ignored."
-            )
+            for position_and_feature_name, feature_value in token_features.items():
+                mapping = self._feature_to_idx_dict.get(position_and_feature_name)
+                if not mapping:
+                    continue
+                feature_idx = mapping.get(feature_value, -1)
+                if feature_idx > -1:
+                    rows.append(token_idx)
+                    cols.append(feature_idx)
+        rows = np.array(rows)
+        cols = np.array(cols)
+        data = np.ones(len(rows))
+        return scipy.sparse.coo_matrix((data, (rows, cols)), shape=shape)
 
-        return value
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> LexicalSyntacticFeaturizerGraphComponent:
+        """Creates a new untrained component (see parent class for full docstring)."""
+        return cls(config, model_storage, resource, execution_context)
 
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Optional[Metadata] = None,
-        cached_component: Optional["LexicalSyntacticFeaturizer"] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "LexicalSyntacticFeaturizer":
+    ) -> LexicalSyntacticFeaturizerGraphComponent:
         """Loads trained component (see parent class for full docstring)."""
-        file_name = meta["file"]
-
-        feature_to_idx_file = Path(model_dir) / f"{file_name}.feature_to_idx_dict.pkl"
-        feature_to_idx_dict = io_utils.json_unpickle(feature_to_idx_file)
-
-        return LexicalSyntacticFeaturizer(meta, feature_to_idx_dict=feature_to_idx_dict)
-
-    def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
-        """Persist this model into the passed directory.
-        Return the metadata necessary to load the model again."""
-
-        feature_to_idx_file = Path(model_dir) / f"{file_name}.feature_to_idx_dict.pkl"
-        io_utils.json_pickle(feature_to_idx_file, self.feature_to_idx_dict)
-
-        return {"file": file_name}
+        try:
+            with model_storage.read_from(resource) as model_path:
+                feature_to_idx_dict = rasa.utils.io.json_unpickle(
+                    model_path / cls.FILENAME_FEATURE_TO_IDX_DICT, keys=True
+                )
+                loaded_featurizer = cls(
+                    config=config,
+                    model_storage=model_storage,
+                    resource=resource,
+                    execution_context=execution_context,
+                )
+                loaded_featurizer._set(
+                    feature_to_idx_dict, check_consistency_with_config=True
+                )
```

Comment:
can't we pass this via the constructor instead of calling an extra method?

Reply:
We could - this is illustrating the separation I meant before. Let's say we train/persist/load multiple attributes (i.e. internal vocabulary + internal model). If we put vocabulary and model into the __init__ both with "Optional" that looks like I could instantiate this component with some model but vocabulary = None which probably makes no sense (otherwise we wouldn't train and persist both). In contrast, this _set (started calling this _load in other PR), emphasises that both need to be set (cause nothing "Optional"). 
Besides, it would also enable us to put a generic "create" into the graph component that just "__init__"s with the same parameters - which means all GraphComponents just need to implement "__init__" as usual (with super().__init__() taking care of graph stuff) and we get the "create" via inheritance.

... I know this is nitpicking but what do you think :D

Reply:
We could also streamline the load() a bit with this... because all classes will do something like:
```
   instance = cls(config, model_storage, execution_context,....)
   try: 
     with  ... context ...:
           # load resource xy 
           # call _set/_load with all loaded resources (and check whether they conflict with the given config)
   except:
         # complain the model cannot be loaded - maybe because it wasn't trained (which is how it currently works for most components)
    return instance
 ```

Reply:
... that said, I can also just put those in the __init__ for now :)

Reply:
> ... I know this is nitpicking but what do you think :D

Haha, In my opinion this is fine. I'd prefer to rather forbid calling the `__init__` directly and always go via `create` / `load` 🤔 I'd go for `__init__` for now to be consistent to how it's done in the other components and then revisit this later.

> because all classes will do something like:

Actually just the ones which train 🤔

Reply:
> Actually just the ones which train 🤔
oh, true - or the ones who "train (but I don't train and just load and persist something downloaded from somehwere)"